src/logical_planner/select.rs

```diff
@@ -135,7 +135,10 @@ impl LogicalPlaner {
         let comparators = stmt.orderby;
 
         // TODO: support the following clauses
-        assert!(!stmt.select_distinct, "TODO: plan distinct");
+        // assert!(!stmt.select_distinct, "TODO: plan distinct");
+        if stmt.select_distinct {
+            plan = Arc::new(LogicalDistinct::new(plan))
```

Comment:
What about directly create `LogicalAgg` here?

Reply:
> What about directly creating `LogicalAgg` here?

LogicalAggregate will create a new column bindings, but LogicalDistinct will not. This is important for `SELECT DISTINCT x FROM test ORDER BY x`. 

LogicalOrderBy is after LogicalDistinct. 

LogicalOrder expr is ColumnRef(x).
LogicalDistinct expr is InputRef(0).
LogicalProjection expr is ColumnRef(x).

Because LogicalDistinct does not generate new bindings, this is no problem, because Order::ColumnRef(x) = Proj::ColumnRef(x), we can successfully create InputRef for LogicalOrder.

But, LogicalAggregate will generate new bindings which causes bindings to become [ InputRef(0) ]
Order::ColumnRef(x)  != Agg::InputRef(0)

Reply:
Sorry but I didn't get your point. IIUC `SELECT DISTINCT x FROM test ORDER BY x` should be equivalent to `SELECT x FROM test GROUP BY x ORDER BY x`, and it seems we can correctly handle the latter one. Why the column binding for Order would become a problem?